### PR TITLE
Feature/bsp for nucleo f303re

### PR DESCRIPTION
**Pull request by luxarf**
I would like to use a Nucleo F303RE with modm and therefore added board support for this development board.

This pull request contains nothing major, the board support files are slightly modified copies of the very similar Nucloe F303K8, corrected for the different form factor.

The examples are slightly modified copies from the examples for the aforementioned Nucleo F303K8 and the Nucleo F103RB.

I verified that the examples are running on the Nucleo F303RE Board that is available to me.

What else needs to be done to make this Pull Request mergeable?

### CODE REVIEW

File: examples/nucleo_f303re/blink/main.cpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2016-2017, Niklas Hauser
```
Comment (rleh) [inline examples/nucleo_f303re/blink/main.cpp:2]: Nope.

File: src/modm/board/nucleo_f303re/board.hpp [hunk 2]
```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2016-2018, Niklas Hauser
+ * Copyright (c) 2017, Sascha Schade
```
Comment (rleh) [inline src/modm/board/nucleo_f303re/board.hpp:3]: No

File: src/modm/board/nucleo_f303re/module.lb [hunk 3]
```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2016-2018, Niklas Hauser
+# Copyright (c) 2017, Fabian Greif
```
Comment (rleh) [inline src/modm/board/nucleo_f303re/module.lb:5]: ...

**Review by rleh — DISMISSED** [review]
Hi @luxarf, thanks for your first contribution to modm!

The pull request looks mostly fine to me!

**Review by salkinium — CHANGES_REQUESTED** [review]
I think the copyright is fine for files you only copied, but the `module.lb` should have your name.
Apart from that, can you squash your commits onto just one?

File: examples/nucleo_f303re/blink/main.cpp (hunk 1 above)
Comment (salkinium) [reply examples/nucleo_f303re/blink/main.cpp:2]: Yope? Isn't this just copied?

File: src/modm/board/nucleo_f303re/board.hpp (hunk 2 above)
Comment (salkinium) [reply src/modm/board/nucleo_f303re/board.hpp:3]: Yes

File: src/modm/board/nucleo_f303re/module.lb (hunk 3 above)
Comment (salkinium) [reply src/modm/board/nucleo_f303re/module.lb:5]: You added some original work here, so proudly add your name 🤓.

**Comment by salkinium** [comment]
(We really need to generate these examples from a common source, this is copy-paste is really getting out of control 🤪🙈)

**Comment by rleh** [comment]
See #359 :wink:

**Comment by luxarf** [comment]
After a little back and forth with git (I am doing a squash for the first time), I am hopeful I got everything.

**Comment by salkinium** [comment]
Hm sorry, I forgot: You also need to add your new examples to the CI in the `.circleci/config.yml`:

```diff
- (cd examples && ../tools/scripts/examples_compile.py stm32f3_discovery nucleo_f303k8)
+ (cd examples && ../tools/scripts/examples_compile.py stm32f3_discovery nucleo_f303k8 nucleo_f303re)
```

I need to write a test that checks that all examples are in the CI, I've also added examples and forgot to add them to the CI. Whoops.

**Comment by luxarf** [comment]
Although the ci checks pass, I am seeing deprecation warnings.
I suspect that stems from the fact that I copied the itm example about a week ago and rebased my branch onto develop before creating this pull-request. Obviously I did not exhibit the necessary care before determining the examples were ready. 
But I am glad the CI catched it. :-)

**Review by salkinium — APPROVED** [review]
Excellent work, thanks!

**Pull request merged**